build-system/tasks/visual-diff/helpers.js

```diff
@@ -48,6 +48,7 @@ function log(mode, ...messages) {
       fancyLog.error(colors.red('ERROR:'), ...messages);
       break;
     case 'fatal':
+      process.exitCode = 1;
```

Comment:
You are correct that the `SIGKILL` change was the fix. This line doesn't affect the stalling, only the exit code (and, as a consequence, whether the rest of the tests in the PR check should be executed)

The rest of this PR is just a bunch of other, related changes I've been wanting to do for a while, to make visual diff tests more robust when it comes to error handling. I didn't like that it's killing the entire execution because of a single error - much like we don't kill the unit/integration tests because of a single error. This was necessary in the past when the tests were slow, but now that they're fast it's not required anymore.